pkgs/applications/editors/emacs-modes/melpa-packages.nix

```diff
@@ -133,11 +133,7 @@ self:
       maxframe = markBroken super.maxframe;
 
       magit =
-        (super.magit.override {
-          # version of magit-popup needs to match magit
-          # https://github.com/magit/magit/issues/3286
-          inherit (self.melpaPackages) magit-popup;
-        }).overrideAttrs (attrs: {
+        super.magit.overrideAttrs (attrs: {
```

Comment:
magit no longer uses magit-popup

Reply:
after the update, magit from elpa would no longer start, complaining, that it didn't find magit-popup. It went back to normal, after I manually installed magit-popup

magit from elpaPackages seemed to have a similar proplem: starting, but being broken, also complaining about absence of magit-popup ...

Reply:
Could you clarify?

1. There is no magit in elpaPackages.
2. The one in melpaStablePackages still has this work-around applied.
3. The one in melpaPackages is running fine for me. (If I revert this diff, I get an evaluation error.)
I left a comment above—magit no longer uses magit-popup, so it must not be passed in here.

Reply:
Regular emacsPackages then, sorry. Here is a [before](https://github.com/bendlas/nixos-config/blob/e8b9d435bda3d95110df14f93498a2c9c92ba47f/emacs.nix) and [after](https://github.com/bendlas/nixos-config/blob/e2032ec30b6e23300837c3ace04882f53f80a3dd/emacs.nix) of my emacs config.

Without `magit-popup`, it fails. I'm on `magit-20190405.1004`

Reply:
Hm, can't reproduce it with a clean config
```
#!/bin/sh

nix-shell --pure -p git "
  with import ./. { config = { }; };
  emacsWithPackages (epkgs: [ epkgs.magit ])
" --run "HOME=/tmp emacs"
```
so it's probably fine. I'll check back, if it's something we can fix ...

Reply:
@rasendubi I found out: it's `magithub`, that still needs `magit-popup`

Reply:
I believe this is an upstream issue.

magithub neither declares `magit-popup` as a dependency nor `(require 'magit-popup)` directly but rather relies on magit to do this.
And magit dropped `magit-popup` from dependencies now.

You might be able to reproduce the issue without Nix infrastructure using package.el: install magit and magithub (magit-popup won't be installed and you'll get the same issue).

If you need a workaround, simply pulling `magit-popup` into scope seems to fix the issue for me.
i.e., the following works fine:
```sh
nix-shell --pure -p git "
  with import ./. { config = { }; };
  emacsWithPackages (epkgs: [ epkgs.magithub epkgs.magit-popup ])
" --run "HOME=/tmp emacs"
```